src/froll.c

```diff
@@ -397,6 +397,277 @@ void frollsumExact(double *x, uint64_t nx, ans_t *ans, int k, double fill, bool
   }
 }
 
+/* fast rolling max */
+void frollmax(unsigned int algo, double *x, uint64_t nx, ans_t *ans, int k, int align, double fill, bool narm, int hasna, bool verbose) {
+  if (nx < k) {
+    if (verbose)
+      snprintf(end(ans->message[0]), 500, _("%s: window width longer than input vector, returning all %f vector\n"), __func__, fill);
+    for (int i=0; i<nx; i++) {
+      ans->dbl_v[i] = fill;
+    }
+    return;
+  }
+  double tic = 0;
+  if (verbose)
+    tic = omp_get_wtime();
+  if (algo==0) {
+    frollmaxFast(x, nx, ans, k, fill, narm, hasna, verbose);
+  } else { // algo==1
+    frollmaxExact(x, nx, ans, k, fill, narm, hasna, verbose);
+  }
+  if (ans->status < 3 && align < 1) {
```

Comment:
`ans->status < 3` doesn't really mean much to me as code reader, how about

```suggestion
if (ans->status < /* error= */ 3 && align < 1) {
```

or

```suggestion
if (ans->status < 3 && align < 1) { // see types.h
```

Reply:
in `frollmean` function this line is commented well: https://github.com/Rdatatable/data.table/blob/0773c44edac01b326e06ecbaffe2914f4103851d/src/froll.c#L29
I don't think it make sense to copy comments to all function when implementation pattern is so similar. As explained before in https://github.com/Rdatatable/data.table/pull/5889/files/63ea485ed24eb3c7ec75f3e57f0f0b4681b76165#r1447373236

Reply:
that could ended up every `if (...) {` line to have comment like `// see frollmean comments for details`

Reply:
Please don't push back on completely reasonable requests for better-documented code like this.

Just one comment at the top of the main entry point to individual rolling functions should be enough. That would be where a new reader could logically be expected to start reading about implementation of individual rolling function.

If that wasn't clear from my comment, please take initiative to come up with such compromises. Completely undocumented code is not acceptable; neither is expecting every new reader of the code to ingest 1000s of lines of C code to figure out what's going on / which secret compartment contains the real comments.

Reply:
added comments in https://github.com/Rdatatable/data.table/pull/5889/commits/03af0e30f1a6a9e75f82b5827c1078f42db48e45

Reply:
Sorry, perhaps my suggestion was not clear enough. Adding a suggestion in a separate comment.

Reply:
ok, added comment as well to frollsum top entry point